lib/jmap/mail/email/email.dart

```diff
@@ -0,0 +1,214 @@
+import 'package:equatable/equatable.dart';
+import 'package:jmap_dart_client/http/converter/body_value_converter.dart';
+import 'package:jmap_dart_client/http/converter/email_id_converter.dart';
+import 'package:jmap_dart_client/http/converter/id_nullable_converter.dart';
+import 'package:jmap_dart_client/http/converter/keyword_identifier_converter.dart';
+import 'package:jmap_dart_client/http/converter/mailbox_ids_converter.dart';
+import 'package:jmap_dart_client/http/converter/message_id_converter.dart';
+import 'package:jmap_dart_client/http/converter/thread_id_nullable_converter.dart';
+import 'package:jmap_dart_client/http/converter/unsigned_int_nullable_converter.dart';
+import 'package:jmap_dart_client/http/converter/utc_date_nullable_converter.dart';
+import 'package:jmap_dart_client/jmap/core/id.dart';
+import 'package:jmap_dart_client/jmap/core/properties/properties.dart';
+import 'package:jmap_dart_client/jmap/core/unsigned_int.dart';
+import 'package:jmap_dart_client/jmap/core/utc_date.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_address.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_body_part.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_body_value.dart';
+import 'package:jmap_dart_client/jmap/mail/email/keyword_identifier.dart';
+import 'package:jmap_dart_client/jmap/mail/mailbox/mailbox.dart';
+
+class Email with EquatableMixin {
+  static Properties allProperties = Properties({
+    'id', 'subject','from', 'to', 'cc', 'bcc', 'keywords', 'size', 'receivedAt',
+    'sentAt', 'replyTo', 'preview', 'hasAttachment'
+  });
+
+  final EmailId id;
+  final Id? blobId;
+  final ThreadId? threadId;
+  final Map<MailboxId, bool>? mailboxIds;
+  final Map<KeyWordIdentifier, bool>? keywords;
+  final UnsignedInt? size;
+  final UTCDate? receivedAt;
+  final Set<MessageId>? messageId;
+  final Set<MessageId>? inReplyTo;
+  final Set<MessageId>? references;
+  final String? subject;
+  final UTCDate? sentAt;
+  final bool? hasAttachment;
+  final String? preview;
+  final Set<EmailAddress>? sender;
+  final Set<EmailAddress>? from;
+  final Set<EmailAddress>? to;
+  final Set<EmailAddress>? cc;
+  final Set<EmailAddress>? bcc;
+  final Set<EmailAddress>? replyTo;
+  final Set<EmailBodyPart>? textBody;
+  final Set<EmailBodyPart>? htmlBody;
+  final Set<EmailBodyPart>? attachments;
+  final EmailBodyPart? bodyStructure;
+  final Map<PartId, EmailBodyValue>? bodyValues;
+
```

Comment:
IMO, you miss `final Set<EmailHeader> headers`

Reply:
you can refer to here: https://github.com/apache/james-project/blob/master/server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailHeader.scala